legend-engine-xt-nonrelationalStore-mongodb-grammar-integration/src/test/java/org/finos/legend/engine/language/pure/grammar/integration/TestMongoDBMappingCompiler.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.language.pure.grammar.integration;
+
+import org.finos.legend.engine.language.pure.compiler.test.TestCompilationFromGrammar;
+import org.junit.Test;
+
+public class TestMongoDBMappingCompiler extends TestCompilationFromGrammar.TestCompilationFromGrammarTestSuite
+{
+
+    @Override
+    protected String getDuplicatedElementTestCode()
+    {
+        return "Class anything::class {}\n" +
+                "###Mapping\n" +
+                "Mapping mongo::test::mapping::MongoDBMapping\n" +
+                "(\n" +
+                "  *meta::external::mongo::mapping::SomeClass[id1]: MongoDB\n" +
+                "  {\n" +
+                "    ~mainCollection [mongo::test::db] PersonRecord\n" +
+                "  }\n" +
+                ")\n" +
+                "###MongoDB\n" +
+                "Database test::mongodb::connection\n" +
+                "(\n" +
+                ")";
+    }
+
+    @Override
+    protected String getDuplicatedElementTestExpectedErrorMessage()
+    {
+        return "COMPILATION error at [3:1-9:1]: Error in 'mongo::test::mapping::MongoDBMapping': Can't find class 'meta::external::mongo::mapping::SomeClass'";
```

Comment:
i think (purely based on the name of the overridden test) that we need two entities with the same name. 
so let us call the ~maincollection & the mapping database having the same qualified name.

Reply:
done